### PR TITLE
LPS-105711 Remove ajax from Liferay.Service

### PR DESCRIPTION
**Pull request by carloslancha**

### CODE REVIEW

**Comment by carloslancha** [comment]
ci:test:sf

**Comment by carloslancha** [comment]
ci:test:relevant

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js [hunk 1]
```diff
@@ -1027,14 +1027,17 @@
...
-				data: {
-					doAsUserId: params.doAsUserId,
-					p_auth: Liferay.authToken,
-					p_l_id: params.plid,
-					portletId: params.portletId,
-					title: params.title
-				}
+			var data = {
+				doAsUserId: params.doAsUserId,
+				p_auth: Liferay.authToken,
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js]: Isn't this already added by our `fetch` wrapper?

File: modules/apps/frontend-js/frontend-js-jquery-web/src/main/java/com/liferay/frontend/js/jquery/web/internal/servlet/taglib/JQueryTopHeadDynamicInclude.java [hunk 2]
```diff
@@ -127,7 +127,7 @@ protected void activate(
 	private static final String[] _FILE_NAMES = {
 		"/jquery/jquery.min.js", "/jquery/bootstrap.bundle.min.js",
 		"/jquery/collapsible_search.js", "/jquery/fm.js", "/jquery/form.js",
-		"/jquery/popper.min.js", "/jquery/side_navigation.js"
+		"/jquery/popper.min.js", "/jquery/side_navigation.js", "/jquery/ajax.js"
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-jquery-web/src/main/java/com/liferay/frontend/js/jquery/web/internal/servlet/taglib/JQueryTopHeadDynamicInclude.java]: I think the only script here that's positional is `jquery.min.js`. The rest, are basically sorted alphabetically. Sort, please? 🙏

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js [hunk 3]
```diff
@@ -207,12 +207,11 @@ Liferay = window.Liferay || {};
...
-			ioConfig.form = form;
-
-			if (ioConfig.form.enctype == STR_MULTIPART) {
-				ioConfig.contentType = false;
-				ioConfig.processData = false;
+			if (form.enctype == STR_MULTIPART) {
+				ioConfig.contentType = 'multipart/form-data';
 			}
+
+			ioConfig.form = new FormData(form);
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js]: Naming here is a bit confusing... if this is `FormData`, why not call this `ioConfig.formData`? It looks like before this was `ioConfig.form.data`?

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js [hunk 4]
```diff
@@ -235,38 +234,24 @@ Liferay = window.Liferay || {};
...
-				$(ioConfig.form)
-					.serializeArray()
-					.forEach(item => {
-						ioConfig.data[item.name] = item.value;
-					});
-			}
-
-			delete ioConfig.form;
+			data = ioConfig.form;
+			data.append('cmd', cmd);
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/liferay.js]: This is the result of my comment above... I was wondering... how's it possible for an HTML form to have an `append` API to add data? Turns out, this is not a form at all, right? :)

**Comment by jbalsas** [comment]
LGTM, @carloslancha, just a couple of minor nits

**Comment by jbalsas** [comment]
See if you want to address them while we wait for the result of the tests

**Comment by carloslancha** [comment]
ci:test:stop

**Comment by carloslancha** [comment]
ci:stop

**Comment by carloslancha** [comment]
ci:test:sf

**Comment by carloslancha** [comment]
ci:test:sf

**Comment by carloslancha** [comment]
ci:test:sf

**Pull request closed without merging**